design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,209 @@
+- When clicked, the field displays the [record selector](#record-selector) component containing a pre-filtered list. A single matching item is displayed.
+- The user can disable the record preview by opening the 'Table Link' preferences from the column header menu and unchecking the 'Show Record Preview' option.
+
+### Scenario 4b: The option for record preview is disabled
+
+#### Steps for 4b
+
+- The user opens a table that has columns with a foreign key constraint applied.
+  - The column contains the primary key value for the referenced column only.
+  - When clicked, the field displays the [record selector](#record-selector) component containing a pre-filtered list. A single matching item is displayed.
+- The user can enable the record preview by opening the 'Table Link' preferences from the column header menu and checking the 'Show Record Preview' option.
+
+## Scenario 5: User Edits the Values of a Column With a Foreign Key Constraint Applied
+
+Linked records can be changed or removed according to the user's preference. Since values are references to records in other tables, users must understand that changes to references don't affect the original records. However, changes to the values of records in the referenced table do.
+
+### Scenario 5a: The field has an existing value
+
+#### Steps for 5a
+
+- The user opens a table that has columns with a foreign key constraint applied.
+  - The column contains the referenced values for each cell.
+  - When clicked, a pre-filtered list is displayed inside the [record selector](#record-selector) component. The dropdown includes a search input, that if modified, will show other matching records. Clicking on any of those records will close the dropdown and replace the existing one.
+  - Note that in the case of multi-column foreign key constraints, the system could modify additional fields.
+
+### Scenario 5b: The field is empty
+
+#### Steps for 5b
+
+- The user opens a table that has columns with a foreign key constraint applied.
+  - The column contains no values.
+  - When clicked, a list of records is displayed inside the [record selector](#record-selector) component. The list shows the first 50 records.
+  - An text input field is available within the dropdown to filter the records based on partial or complete value matches. As the user types in a value, the list narrows down the records. The system can search only the first five fields. Search across all types is supported.
+  - If there's a single match for the entered value, it should become highlighted, allowing the user to confirm the selection.
+  - If the table contains no records, the user sees an option to add a record.
+  - If there aren't any matches, the user sees an option to add a record for the newly entered value or try changing the 'Search Columns' table preference.
```

Comment:
What happens after the user clicks "Add Record" here?

Reply:
@seancolsen we haven't defined that part of the flow yet. I assume we'll know more when we work on views as it also involves the creation of new records in other tables. I'll add a note for this.

Reply:
I think we should remove this option from the spec, we can add it in once we have it defined. It'll confuse implementation.